docs/references/glossary.md

```diff
@@ -0,0 +1,328 @@
+
+## Component (catalog plugin)
+
+A software product that is managed in the Backstage [Software Catalog](#software-catalog). A component can be a service, website, library, data pipeline, or any other piece of software managed as a single project.
+
+https://backstage.io/docs/features/software-catalog/system-model
+
+## Condition (permission plugin)
+
+Conditions are used to return a conditional decision from a policy. They contain information about a given entity and restrictions on what types of users can view that entity.
+
+## Conditional decisions
+
+[Rules](#permission-rules) need additional data before they can be used in a decision. Once a [rule](#permission-rule) is bound to relevant information it forms a [condition](#condition). Conditional decisions tell the [permission framework](#permission) to delegate evaluation to the [plugin](#plugin) that owns the corresponding [resource](#permission-resource). Permission requests that result in a conditional decision are allowed if all of the provided conditions evaluate to be true.
+
+## Contributor
+
+A volunteer who helps to improve an OSS product such as Backstage. This volunteer effort includes coding, testing, technical writing, user support, and other work. A [user role](#user-role).
+
+## Declarative integration
+
+A new paradigm for Backstage frontend plugins, allowing definition in config files instead of hosting complete React pages.
+
+https://backstage.io/docs/frontend-system
+
+## Decorators (search plugin)
+
+A transform stream. Decorators allow you to add additional information to documents outside of the [collator](#collators). They sit between the [collators](#collators) and the [indexers](#indexer) and can add extra fields to documents as they're being collated and indexed.
+
+Possible use cases for a decorator could be to bias search results or otherwise improve the search experience in your Backstage instance. Decorators can also be used to remove [metadata](#metadata), filter out, or even add extra documents at index-time. Part of [Backstage Search](#search).
+
+## Deployment Artifacts
+
+An executable or package file with all of the necessary information required to deploy the application at runtime. Deployment artifacts can be hosted on [package registries](#package-registry).
+
+## Developer
+
+Someone who writes code and develops software.
+
+A [user role](#user-role) defined as someone who uses a Backstage [app](#app). Might or might not actually be a software developer.
+
+## Developer Portal
+
+A centralized system comprising a user interface and database used to facilitate and document all the software projects within an organization. Backstage is both a developer portal and (by virtue of being based on plugins) a platform for creating developer portals.
+
+## Documents (search plugin)
```

Comment:
```suggestion
## Document (search plugin)
```

Reply:
By convention, glossary entries are singular. But, need to change references too if you do this. Might not be worth it.

Reply:
It's worth the time in the first round to get this right, updated in https://github.com/backstage/backstage/pull/22542/commits/4944f29f2cc61dbd29019cd5693c051301f8497f